### PR TITLE
Fix the candidates and user tables date bug

### PR DESCRIPTION
**Pull request by 123MwanjeMike**
#### Summary of Changes (What does this PR do?)
- Fixes the candidates and user tables date bug

#### Status of maturity (all need to be checked before merging):
- [x] I've tested this locally
- [x] I consider this code done
- [x] This change ready to hit production in its current state
- [x] The title of the PR states what changed and the related issues number (used for the release note).
- [ ] I've included issue number in the "Closes #ISSUE-NUMBER" part of the "[What are the relevant tickets?](#what-are-the-relevant-tickets)" section to [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [ ] I've updated corresponding documentation for the changes in this PR.
- [ ] I have written unit and/or e2e tests for my change(s).

#### How should this be manually tested?
- Checkout https://fix-candidatesdatesbug-netmanager-preview-w7kzhvlewq-ew.a.run.app/candidates
- You should see records without dates having "---"

#### Screenshots (optional)
image

image

### CODE REVIEW

File: netmanager/src/views/pages/CandidateList/components/CandidatesTable/CandidatesTable.js [hunk 1]
```diff
@@ -262,7 +262,9 @@ const CandidatesTable = (props) => {
           {
             title: 'Submitted',
             field: 'createdAt',
-            render: (candidate) => <span>{formatDateString(candidate.createdAt)}</span>
+            render: (candidate) => (
+              <span>{candidate.createdAt ? formatDateString(candidate.createdAt) : '---'}</span>
```
Comment (123MwanjeMike) [inline netmanager/src/views/pages/CandidateList/components/CandidatesTable/CandidatesTable.js:266]: require createdAt field before formating string

File: netmanager/src/views/pages/UserList/components/UsersTable/UsersTable.js [hunk 2]
```diff
@@ -148,225 +147,213 @@ const UsersTable = (props) => {
...
+          {
+            title: 'Role',
+            field: 'privilege'
+          },
+          {
+            title: 'Joined',
+            field: 'createdAt',
+            render: (candidate) => (
+              <span>{candidate.createdAt ? formatDateString(candidate.createdAt) : '---'}</span>
+            )
```
Comment (123MwanjeMike) [inline netmanager/src/views/pages/UserList/components/UsersTable/UsersTable.js:188]: require createdAt field before formating string

Comment (Codebmk) [reply netmanager/src/views/pages/UserList/components/UsersTable/UsersTable.js:188]: @123MwanjeMike the createdAt field is automatically added into the db when a new candidate registers so adding the ternary operator is not needed. Just add `{formatDateString(candidate.createdAt)}` except if you found instances of empty fields?

Comment (123MwanjeMike) [reply netmanager/src/views/pages/UserList/components/UsersTable/UsersTable.js:188]: Some records didn't have it. Which caused @wabinyai's bug that @Baalmart shared via mail on Wednesday.

Comment (Codebmk) [reply netmanager/src/views/pages/UserList/components/UsersTable/UsersTable.js:188]: Oh okay

**Review by Codebmk — APPROVED** [review]
LGTM thanks @123MwanjeMike

**Review by Baalmart — APPROVED** [review]
LGTM

**Pull request merged**